lib/irc/Scheduler.js

```diff
@@ -0,0 +1,64 @@
+/*eslint no-invalid-this: 0*/
+"use strict"
+var Promise = require("bluebird");
+var logging = require("../logging");
+var log = logging.get("scheduler");
+var Queue = require("../util/Queue.js");
+
+/**
+ * An IRC connection scheduler. Enables ConnectionInstance to reconnect
+ * in a way that queues reconnection requests and services the FIFO queue at a
+ * rate determined by ircServer.getReconnectIntervalMs().
+ */
+
+ var queues = {};
```

Comment:
In-line comment explaining what this is would be super. e.g `// domain => Queue`